pkg/helm/controller.go

```diff
@@ -150,21 +166,37 @@ func (c *Controller) OnHelmRemove(key string, chart *helmv1.HelmChart) (*helmv1.
 	return newChart, c.apply.WithOwner(newChart).Apply(objectset.NewObjectSet())
 }
 
+func (c *Controller) OnConfChange(key string, conf *helmv1.HelmChartConfig) (*helmv1.HelmChartConfig, error) {
+	if conf == nil {
+		return nil, nil
+	}
+
+	if chart, err := c.helmController.Cache().Get(conf.Namespace, conf.Name); err != nil {
+		if !errors.IsNotFound(err) {
+			return conf, err
+		}
+	} else if chart != nil {
+		if _, err := c.OnHelmChange(key, chart); err != nil && err != apply.ErrReplace {
```

Comment:
Hm. In this case, I've seen a pattern where we actually just enqueue the other object (in this case the HelmChart) rather than calling the change handler directly from another handler.

Reply:
Don't invoke OnHelmChange directly, IMO, enqueue it instead.